File: python/stack.py
from linked_list import *

class Stack(object):
    def __init__(self,top=None):
        self.ll = LinkedList(top)

    def push(self, new_element):
        self.ll.insert_first(new_element)
        pass

    def pop(self):
        return self.ll.delete_first()

File: ruby/stack/stack.rb
require_relative '../linked_list/linked_list'

class Stack
  def initialize(head = nil)
    @ll = LinkedList.new(head)
  end

  def push(new_element)
    @ll.insert_first(new_element)
  end

  def pop
    @ll.delete_first
  end
end
